packages/expression-evaluator/test/integration/evaluators/TermComparator-test.ts

```diff
@@ -0,0 +1,184 @@
+  typeDiscoveryCallback?: SuperTypeCallback) {
+  await orderTestIsLower(litA, litB, typeDiscoveryCallback);
+}
+
+async function orderTestIsEqual(litA: RDF.Term | undefined, litB: RDF.Term | undefined) {
+  const evaluator = await getMockEEFactory()
+    .createTermComparator({
+      context: getMockEEActionContext(),
+    });
+  expect(evaluator.orderTypes(litA, litB)).toEqual(0);
+  expect(evaluator.orderTypes(litB, litA)).toEqual(0);
+}
+
+describe('terms order', () => {
+  it('undefined is equal to undefined', async() => {
+    await orderTestIsEqual(undefined, undefined); // eslint-disable-line unicorn/no-useless-undefined
+  });
+
+  it('undefined is lower than everything else', async() => {
+    await genericOrderTestLower(undefined, DF.blankNode());
+    await genericOrderTestLower(undefined, DF.namedNode('http://example.com'));
+    await genericOrderTestLower(undefined, DF.literal('foo'));
+  });
+
+  it('blank nodes are ordered based on their ids', async() => {
+    await orderTestIsEqual(DF.blankNode('a'), DF.blankNode('a'));
+    await genericOrderTestLower(DF.blankNode('a'), DF.blankNode('b'));
+  });
+
+  it('blank nodes are lower than other terms', async() => {
+    await genericOrderTestLower(DF.blankNode(), DF.namedNode('http://example.com'));
+    await genericOrderTestLower(DF.blankNode(), DF.literal('foo'));
+  });
+
+  it('named nodes are ordered based on their IRI strings', async() => {
+    await orderTestIsEqual(DF.namedNode('http://example.com/a'), DF.namedNode('http://example.com/a'));
+    await genericOrderTestLower(DF.namedNode('http://example.com/a'), DF.namedNode('http://example.com/b'));
+  });
+
+  it('named nodes are lower than literals', async() => {
+    await genericOrderTestLower(DF.namedNode('http://example.com'), DF.literal('foo'));
+  });
+
+  it('integers type identical', async() => {
+    await orderTestIsEqual(int('11'), int('11'));
+  });
+
+  it('string type identical', async() => {
+    await orderTestIsEqual(string('11'), string('11'));
+  });
+
+  it('string type comparison', async() => {
+    await genericOrderTestLower(string('11'), string('2'));
+  });
+  it('integer type comparison', async() => {
+    await genericOrderTestLower(int('2'), int('11'));
+  });
+  it('double type comparison', async() => {
+    await genericOrderTestLower(double('2'), double('11'));
+  });
+  it('decimal type comparison', async() => {
+    await genericOrderTestLower(decimal('2'), decimal('11'));
+  });
+  it('float type comparison', async() => {
+    await genericOrderTestLower(float('2'), float('11'));
+  });
+  it('dateTime type comparison', async() => {
+    await genericOrderTestLower(dateTime('2000-01-01T00:00:00Z'), dateTime('2001-01-01T00:00:00Z'));
+  });
+  it('langString type comparison', async() => {
+    // Spec does not say anything about order of langStrings, but we use Operator Extensibility to define it.
```

Comment:
This is a false note! We do not have any operator extensibility in place for languageString `<`! 
I wonder whether we want this? @rubensworks What do you think? Right now (and since a long time), this test passes by accident.

Reply:
I would keep the same behaviour as we have now as introduced in https://github.com/comunica/sparqlee/pull/139
Do you think this is reasonable?

Reply:
Don't we want ORDER BY to be equivalent to the `<` operator?
Or do you have a reason in mind why we might not want that?

Reply:
I'll leave the choice up to you, as long as no current behaviour is broken :-)

Reply:
Okay! Then I won't implement it since I don't think it makes sense in `<`. And for the order by case, there are tests in place, so that should be alright too. Thank you!